test/unit/spec/IDPDiscovery_spec.js

```diff
@@ -0,0 +1,1031 @@
+        });
+      });
+      itp('has a username field', function () {
+        return setup().then(function (test) {
+          var username = test.form.usernameField();
+          expect(username.length).toBe(1);
+          expect(username.attr('type')).toEqual('text');
+          expect(username.attr('id')).toEqual('idp-discovery-username');
+        });
+      });
+      itp('has a sign in button', function () {
+        return setup().then(function (test) {
+          var signInButton = test.form.signInButton();
+          expect(signInButton.length).toBe(1);
+          expect(signInButton.attr('type')).toEqual('submit');
+          expect(signInButton.attr('id')).toEqual('idp-discovery-submit');
+        });
+      });
+      itp('has a rememberMe checkbox if features.rememberMe is true', function () {
+        return setup().then(function (test) {
+          var cb = test.form.rememberMeCheckbox();
+          expect(cb.length).toBe(1);
+        });
+      });
+      itp('does not have a rememberMe checkbox if features.rememberMe is false', function () {
+        return setup({ 'features.rememberMe': false }).then(function (test) {
+          var cb = test.form.rememberMeCheckbox();
+          expect(cb.length).toBe(0);
+        });
+      });
+      itp('has "Need help?" link', function () {
+        return setup().then(function (test) {
+          expect(test.form.helpFooterLabel().trim()).toBe('Need help signing in?');
+        });
+      });
+      itp('has a help link', function () {
+        return setup().then(function (test) {
+          expect(test.form.helpLinkLabel().trim()).toBe('Help');
+        });
+      });
+      itp('has the correct help link url', function () {
+        return setup().then(function (test) {
+          spyOn(SharedUtil, 'redirect');
+          expect(test.form.helpLinkHref()).toBe('https://foo.com/help/login');
+        });
+      });
+      itp('has a custom help link url when available', function () {
+        return setup({ 'helpLinks.help': 'https://bar.com' }).then(function (test) {
+          spyOn(SharedUtil, 'redirect');
+          expect(test.form.helpLinkHref()).toBe('https://bar.com');
+        });
+      });
+      itp('has a forgot password link', function () {
+        return setup().then(function (test) {
+          expect(test.form.forgotPasswordLabel().trim()).toBe('Forgot password?');
+        });
+      });
+      itp('forgot password link is not visible on load', function () {
+        return setup().then(function (test) {
+          expect(test.form.forgotPasswordLinkVisible()).toBe(false);
+        });
+      });
+      itp('shows forgot password link when clicking help', function () {
+        return setup().then(function (test) {
+          test.form.helpFooter().click();
+          expect(test.form.forgotPasswordLinkVisible()).toBe(true);
+        });
+      });
+      itp('does not show forgot password link when disabled and clicked', function () {
```

Comment:
I don't entirely understand this test. It sets username, clicks submit and then it tries to click the help link?
My question is: after clicking submit on the form and getting a success response, the help link is gone, right?

Reply:
The help link is disabled after clicking submit. So this is testing that clicking on a disabled help link does not display the forgot password link. The majority of these tests, including this one, are from the primary auth spec. I wanted to make sure i wasn't breaking any existing behavior.